kubeflow/core/tensorboard.libsonnet

```diff
@@ -0,0 +1,82 @@
+{
+	parts(namespace, name,):: {
+		service:: {
+		  "apiVersion": "v1", 
+		  "kind": "Service", 
+		  "metadata": {
+		    "name": name + "-tb",
+		    "namespace": namespace,
+		  }, 
+		  "spec": {
+		    "ports": [
+		      {
+		        "name": "http", 
+		        "port": 80, 
+		        "targetPort": 80,
+		      }
+		    ], 
+		    "selector": {
+		      "app": "tensorboard",
+		      "tb-job": name,
+		    },
+		  },
+		},
+
+		tbDeployment(logDir, secretName, secretFileName, tfImage="gcr.io/tensorflow/tensorflow:latest"):: {
+		  "apiVersion": "apps/v1beta1", 
+		  "kind": "Deployment", 
+		  "metadata": {
+		    "name": name + "-tb",
+		    "namespace": namespace,
+		  }, 
+		  "spec": {
+		    "replicas": 1, 
+		    "template": {
+		      "metadata": {
+		        "labels": {
+		          "app": "tensorboard",
+		          "tb-job": name,
+		        }, 
+		        "name": name,
+		        "namespace": namespace,
+		      }, 
+		      "spec": {
+		        "containers": [
+		          {
+		            "command": [
+		              "/usr/local/bin/tensorboard", 
+		              "--logdir=" + logDir, 
+		              "--port=80"
+		            ], 
+		            "image": tfImage, 
+		            "name": "tensorboard", 
+		            "ports": [
+		              {
+		                "containerPort": 80
+		              }
+		            ],
+		            "env":   [
+			           {
+			           "name": "GOOGLE_APPLICATION_CREDENTIALS",
+			           "value": "/secret/gcp-credentials/" + secretFileName,
```

Comment:
Sorry, but is this expecting GCP credentials to run?

Reply:
@abkosar Take a look at what we do for TFServing
https://github.com/kubeflow/kubeflow/blob/master/kubeflow/tf-serving/tf-serving.libsonnet#L253

In fact it probably makes sense just to copy it and then change the needed values like
  - image
  - container args.

That config file shows how we can add environment variables and secrets as needed to create credentials to support S3 or GCS access.